api/opds.py

```diff
@@ -286,14 +300,67 @@ def annotate_work_entry(self, work, active_license_pool, edition, identifier, fe
             )
         )
 
-    @classmethod
-    def related_books_available(cls, license_pool):
-        """:return: bool asserting whether related books are available for a
-        particular work
-        """
-        contributions = license_pool.presentation_edition.contributions
-        series = license_pool.presentation_edition.series
-        return contributions or series or NoveListAPI.is_configured()
+    def language_and_audience_key_from_work(self, work):
+        language_key = work.language
+
+        if work.audience == Classifier.AUDIENCE_CHILDREN:
+            audiences = [Classifier.AUDIENCE_CHILDREN]
+        if work.audience == Classifier.AUDIENCE_YOUNG_ADULT:
+            audiences = Classifier.AUDIENCES_JUVENILE
+        if work.audience in Classifier.AUDIENCES_ADULT:
+            audiences = list(Classifier.AUDIENCES)
+
+        audience_key=None
+        if audiences:
+            audience_strings = [urllib.quote_plus(a) for a in sorted(audiences)]
+            audience_key = u','.join(audience_strings)
+
+        return language_key, audience_key
+
+    def add_author_links(self, work, feed, entry):
+        author_tag = '{%s}author' % OPDSFeed.ATOM_NS
+        author_entries = entry.findall(author_tag)
```

Comment:
As per my suggestion in your core branch, you would also need to do this for the atom:contributor tag.